jena-kafka-connector/src/main/java/org/apache/jena/kafka/KConnectorDesc.java

```diff
@@ -95,18 +118,165 @@ public String getConsumerGroupId() {
+    /**
+     * Gets the number of batch sizes to track in order to average batch sizes over this and determine whether we're
+     * connected to low volume topics.
+     * <p>
+     * Once at least this many batches have been processed if the average batch size falls below the configured
+     * {@link #getLowVolumeBatchSizeThreshold()} then the {@link org.apache.jena.kafka.common.FusekiProjector} switches
+     * over to low volume batching mode.  In this mode it does not automatically commit as soon as it reaches lag of 0
+     * instead waiting until either the batch size, or the {@link #getMaxTransactionDuration()} is exceeded.
+     * </p>
+     * <p>
+     * May be configured by setting the custom Fuseki Kafka configuration property
+     * {@value SysJenaKafka#FUSEKI_KAFKA_BATCH_SIZE_TRACKING_WINDOW}. Default is
+     * {@value SysJenaKafka#DEFAULT_BATCH_SIZE_TRACKING_WINDOW} batches.
+     * </p>
+     *
+     * @return Number of batch sizes to track
+     */
+    public int getBatchSizeTrackingWindow() {
+        return fromKafkaProperties(SysJenaKafka.FUSEKI_KAFKA_BATCH_SIZE_TRACKING_WINDOW, Integer::parseInt,
+                                   IS_POSITIVE_INTEGER, SysJenaKafka.DEFAULT_BATCH_SIZE_TRACKING_WINDOW);
+    }
+
+    /**
+     * Gets the low volume batch size threshold
+     * <p>
+     * If the average batch size over the last {@link #getBatchSizeTrackingWindow()} batches is less than, or equal to,
+     * this threshold then the {@link org.apache.jena.kafka.common.FusekiProjector} switches over to low volume batching
+     * mode.  See {@link #getBatchSizeTrackingWindow()} for more details on this.
+     * </p>
+     * <p>
+     * May be configured by setting the custom Fuseki Kafka configuration property
+     * {@value SysJenaKafka#FUSEKI_KAFKA_LOW_VOLUME_THRESHOLD} in the consumer properties provided to this connector.
+     * Defaults to {@value SysJenaKafka#DEFAULT_AVERAGE_BATCH_SIZE_LOW_VOLUME_THRESHOLD}.
+     * </p>
+     *
+     * @return Low volume batch size threshold
+     */
+    public int getLowVolumeBatchSizeThreshold() {
+        return fromKafkaProperties(SysJenaKafka.FUSEKI_KAFKA_LOW_VOLUME_THRESHOLD, Integer::parseInt, x -> x >= 0,
+                                   SysJenaKafka.DEFAULT_AVERAGE_BATCH_SIZE_LOW_VOLUME_THRESHOLD);
+    }
+
+    /**
+     * Gets the high lag threshold, in number of events.
+     * <p>
+     * If the lag exceeds this then the {@link org.apache.jena.kafka.common.FusekiProjector} switches to high lag
+     * batching mode.  In this mode it no longer honours the {@link #getBatchSize()} instead only committing once
+     * {@link #getBatchSizeBytes()} has been reached.
+     * </p>
+     * <p>
+     * May be configured by setting the custom Fuseki Kafka configuration property
+     * {@value SysJenaKafka#FUSEKI_KAFKA_HIGH_LAG_THRESHOLD} in the consumer properties provided to this connector.
+     * Default is {@value SysJenaKafka#DEFAULT_HIGH_LAG_BATCH_BYTE_THRESHOLD} bytes.
```

Comment:
```suggestion
     * Default is {@value SysJenaKafka#DEFAULT_HIGH_LAG_BATCH_BYTE_THRESHOLD}.
```

Painfully pedantic - It's not bytes but a count